src/cmd-init

```diff
@@ -95,10 +95,7 @@ mkdir -p src
  if ! test -e config; then
      case "${source}" in
          /*) ln -s "${source}/${subdir}" config;;
-         *) git clone --recurse-submodules "${source}" config
-            if [ -n "${BRANCH}" ]; then
-                git -C config checkout "${BRANCH}"
-            fi
+         *) git clone "${BRANCH:+--branch=${BRANCH}}" --recurse-submodules "${source}" config
```

Comment:
I think we want to drop the quotes here so that it evaluates to no arguments if there are no branches?

Reply:
Good catch!  Fixed.